stix2validator/validator.py

```diff
@@ -623,5 +621,10 @@ def validate_instance(instance, options=None):
     else:
         valid = True
 
-    return ValidationResults(is_valid=valid, errors=error_list,
-                             warnings=warnings)
+    # Clear requests cache if commandline flag was set
+    if options.clear_cache:
+        now = datetime.datetime.utcnow()
+        requests_cache.get_cache().remove_old_entries(now)
```

Comment:
Should this caching stuff really be redone for every object?  It seems like it should be set up at the beginning of the commandline script run (say), and then all object validation should be done, and then the cache cleared.

Reply:
No, it should be per-command-line execution (I think). I was under the impression that the "clear cache" was intended to be done *before* validation, but I could be wrong about that, based on what you said in the commit message.

Reply:
The config code for it is:
```python
    parser.add_argument(
        "--clear-cache",
        dest="clear_cache",
        action="store_true",
        default=False,
        help="Clear the cache of external source values after validation."
    )
```
So it seems pretty clear what the author thought.  The probable confusion was that most of the code in `validate_instance()` collects _generators_ together, which will perform the validation when iterated through.  It looks like those function calls are doing validation, but they aren't.  I actually updated the docstring on `_schema_validate()` to make this clearer.  The iteration doesn't happen until the loop that collects the error strings.  So the cache clearing would have to happen after that, for it to occur after validation.

But it sounds like that needs to be moved elsewhere anyway, so it's not cleared for each object.

Reply:
@chisholm is correct. `--refresh-cache` is the option for clearing _before_ validation. Yeah, it should get moved.